hkmc2/shared/src/test/mlscript/basics/MultiParamListClasses.mls

```diff
@@ -0,0 +1,89 @@
+:js
+
+
+data class Foo(x, y)(z)
+
+Foo
+//│ = [function Foo]
+
+Foo(1, 2)
+//│ = [function]
+
+Foo(1, 2)(3)
+//│ = Foo(1, 2)
+
+
+new Foo(1, 2)
+//│ = [function]
+
+:fixme
+new Foo(1, 2)(3)
+//│ ═══[RUNTIME ERROR] TypeError: tmp1 is not a constructor
```

Comment:
The current JS trick to implement multiple parameter lists for classes does not work for `new` operator.

Reply:
Actually it does, it's just that our parsing of `new` is currently broken (any fix welcome; in any case this can be done separately).

```fs
(new Foo(1, 2))(3)
//│ = Foo(1, 2)
```